test/Microsoft.ML.Predictor.Tests/TestTrainerEstimators.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.Runtime.RunTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class SimpleEstimatorTests : TestDataPipeBase
+    {
+        private IDataView GetBreastCancerDataviewWithTextColumns()
+        {
+            var dataPath = GetDataPath("breast-cancer.txt");
+            var inputFile = new SimpleFileHandle(Env, dataPath, false, false);
+            return ImportTextData.TextLoader(Env, new ImportTextData.LoaderInput()
+            {
+                Arguments =
+                {
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoader.Column("Label", type: null, 0),
+                        new TextLoader.Column("F1", DataKind.Text, 1),
+                        new TextLoader.Column("F2", DataKind.I4, 2),
+                        new TextLoader.Column("Rest", type: null, new [] { new TextLoader.Range(3, 9) })
+                    }
+                },
+
+                InputFile = inputFile
+            }).Data;
+        }
+
+        public SimpleEstimatorTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void TestEstimatorRandom()
+        {
+            var dataView = GetBreastCancerDataviewWithTextColumns();
+            var pipe = new RandomTrainer(Env, new RandomTrainer.Arguments());
+
+            // Test only that the schema propagation works.
+            // REVIEW: the save/load is not preserving the full state of the random predictor. This is unfortunate, but we don't care too much at this point.
```

Comment:
>REVIEW: the save/l [](start = 15, length = 18)

maybe log an issue after merging

Reply:
Ok will do that!

---
In reply to: [218139123](https://github.com/dotnet/machinelearning/pull/876#discussion_r218139123) [](ancestors = 218139123)